routing/DELEGATED_ROUTING_HTTP.md

```diff
@@ -0,0 +1,101 @@
+# ![](https://img.shields.io/badge/status-wip-orange.svg?style=flat-square) Delegated Routing HTTP API
+
+**Author(s)**:
+- Gus Eggert
+
+**Maintainer(s)**:
+
+* * *
+
+**Abstract**
+
+"Delegated routing" is a mechanism for IPFS implementations to use for offloading content routing to another process/server. This spec describes an HTTP API for delegated routing.
+
+# Organization of this document
+
+- [Introduction](#introduction)
+- [Spec](#spec)
+  - [Interaction Pattern](#interaction-pattern)
+  - [Cachability](#cachability)
+  - [Transports](#transports)
+  - [Protocol Message Overview](#protocol-message-overview)
+    - [Known Methods](#known-methods)
+- [Method Upgrade Paths](#method-upgrade-paths)
+- [Implementations](#implementations)
+
+# API Specification
+By default, the Delegated Routing HTTP API uses the `application/json` content type. Clients and servers may optionally negotiate other content types such as `application/cbor`, `application/vnd.ipfs.rpc+dag-json`, etc. using the standard `Accept` and `Content-Type` headers.
+
+- `GET /v1/providers/{CID}`
+    - Reframe equivalent: FindProviders
+    - Response
+        
+        ```json
+        {
+        	"Providers": [
+        		{
+        			"PeerID": "...",
+        			"Multiaddrs": ["...", "..."]
+        			"Protocols": [
+        				{
+        					"Codec": 2320,
+        					"Payload": <opaque data>
+        				}
+        			]
+        		}
+        	]
+        	"NextPageToken": "<token>"
+        }
+        ```
+        
+    - Default limit: 100 providers
+    - Optional query parameters
+        - `transfer` only return providers who support the passed transfer protocols, expressed as a comma-separated list of multicodec IDs such as `2304,2320`,
```

Comment:
Can we use the name here? It will be easier to use and remember the common cases.

Reply:
I would love that, but sadly there is no canonical mapping from name to multicodec code. Which I think would a bunch of complexity to consumers since they'll have to do this mapping themselves. I think it'd be easier on folks to just use the code here.

Reply:
:cry: